src/adaptors.rs

```diff
@@ -77,6 +77,17 @@ impl<I, J> Iterator for Interleave<I, J>
             }
         }
     }
+
+    fn size_hint(&self) -> size_hint::SizeHint {
```

Comment:
This should be the literal type, not SizeHint. Since the latter is for internal use only.